Improve documentation (#15)
PhilanthropyDataCommons/sdk: README.md

This repository houses the infrastructure for generating PDC SDKs.

> We're using a project called [swagger-codegen](https://github.com/swagger-api/swagger-codegen) to power our SDK, and this repository provides tooling and templates to make sdk generation easier in specifically supported languages.

Reviewer: ```suggestion
We're using a project called [swagger-codegen](https://github.com/swagger-api/swagger-codegen) to power our SDK, and this repository provides tooling and templates to make SDK generation easier in specifically supported languages.
```

## Prequisites

- JRE (18+)

## Generating SDKs

> To generate the SDKs supported by this project, you will need a [swagger](https://swagger.io/) specification located at the project root and named `openapi.json`. The current swagger specification for the PDC can be found in the [service repository](https://github.com/PhilanthropyDataCommons/service/blob/main/src/openapi.json).

Reviewer: ```suggestion
To generate the SDKs supported by this project, you will need a [Swagger](https://swagger.io/) specification located at the project root and named `openapi.json`. The current Swagger specification for the PDC can be found in the [service repository](https://github.com/PhilanthropyDataCommons/service/blob/main/src/openapi.json).
```

### Generating official SDKs:

1. Clone the repository:

```shell
git clone git@github.com:PhilanthropyDataCommons/sdk.git
```

2. Change directory:

```shell
cd sdk
```

> 3. Copy your swagger specification

Reviewer: ```suggestion
3. Copy your Swagger specification:
```

Colon added for consistency w/ steps 1 and 2.

```shell
cp YOUR_SWAGGER_SPEC_LOCATION ./openapi.json
```

> 4. Generate the code bases

Reviewer: ```suggestion
4. Generate the codebases:
```

Author: <img width="262" alt="image" src="https://github.com/PhilanthropyDataCommons/sdk/assets/208884/3059db51-3c45-4179-a13e-12bb9e2f60bf">

:-|

Author: I decided to ditch "codebase" or "code base" entirely

Reviewer: …weird. I'd say that's a dictionary with insufficient domain knowledge!

FWIW Wikipedia implies both are correct — "codebase (or code base)" — but goes on to use "codebase" throughout the article: https://en.wikipedia.org/wiki/Codebase

Author: yeah the consensus is all of it is ok (including `code-base`) but I just realized the word is longer than needed anyway, I'm not even sure it's the correct term in this case.

(the project isn't swagger-codebasegen it's swagger-codegen!)

```shell
> ./gradlew generateSwaggerCode
```

The generated code will now exist in the `build` directory.

> Each language target will have its own directory inside of `build`, and you can follow the readme instructions in that directory with additional steps related to that code base.

Reviewer: ```suggestion
Each language target will have its own directory inside of `build/`, and you can follow the readme instructions in that directory with additional steps related to that codebase.
```

Including the trailing slash is a convention I've adopted (and like!) when discussing directories.

> - Typescript: `build/typescript/README.md`.

Reviewer: Dan you're going to laugh but I genuinely think this line would be better without the period at the end

Author: I agree!  BUT DID NOT LAUGH.

Reviewer: Dan how many tickles does it take to make an octopus laugh

> - Typescript: `build/typescript/README.md`.

Reviewer: ```suggestion
- TypeScript: `build/typescript/README.md`
```

#### Steps:

1. Clone the repository:

```shell
git clone git@github.com:PhilanthropyDataCommons/sdk.git
```

2. Change directory:

```shell
cd sdk
```

3. Copy your swagger specification

```shell
cp YOUR_SWAGGER_SPEC_LOCATION ./openapi.json
```

4. Generate the code bases

```shell
> ./gradlew generateSwaggerCode
```

The generated code will now exist in the `build` directory.

> Each language target will have its own directory inside of `build`, and you can follow the readme instructions in that directory with additional steps related to that code base.

Reviewer: Should add in generic instructions here (at least one sentence / a link to their docs on how to use codegen without using our custom templates)

Author: oh this is a great point

(Also whoops -- I had local changes I apparently hadn't pushed before requesting review!  Please check your inbox for an email asking you to re-review).

Reviewer: I _only_ make great points.
